### PR TITLE
refactor(substrait): refactor ReadRel consumer

### PR DESCRIPTION
**Pull request by tokoko**
## Rationale for this change

some minor refactoring after #12800 as promised.

## Are these changes tested?

existing tests

## Are there any user-facing changes?

no

### CODE REVIEW

**Review by alamb — APPROVED** [review]
Looks like an improvement to me -- thank you @tokoko

**Review by vbarua — APPROVED** [review]
One minor comment, but overall this looks good to me. Thanks for following up with the 🧹

File: datafusion/substrait/src/logical_plan/consumer.rs [hunk 1]
```diff
@@ -800,79 +801,71 @@ pub async fn from_substrait_rel(
...
-                    substrait_datafusion_err!("No base schema provided for LocalFiles")
-                })?;
-
-                fn extract_filename(name: &str) -> Option<String> {
-                    let corrected_url =
-                        if name.starts_with("file://") && !name.starts_with("file:///") {
+                    Ok(LogicalPlan::Values(Values {
+                        schema: DFSchemaRef::new(substrait_schema),
+                        values,
+                    }))
```
[the remark this answers is not in the document]
Comment (tokoko) [reply datafusion/substrait/src/logical_plan/consumer.rs:823]: That's a good point, but I think the implementation will have to be different. `NamedTable` and `LocalFiles` translate to `TableScan` operator which has a `projections` field that is used by `read_with_schema` function. `VirtualTable` translates to `Values` operator instead that doesn't have a projections field, we will have to apply a Project on top. I'll open a ticket to track this.

**Comment by alamb** [comment]
🚀

**Pull request merged**